src/models/audio.py

```diff
@@ -0,0 +1,79 @@
+import torch
+import numpy as np
+from typing import List, Union
+from .utils import get_mask, masked_emb_pooling
+from transformers import WhisperModel, WhisperFeatureExtractor, AutoProcessor, SeamlessM4TModel
+
+
+class BaseAudioEncoder(torch.nn.Module):
+    def __init__(
+            self,
+            proj_dim: int,
+            device: str = "cpu"
+    ):
+        super().__init__()
+        self.proj_dim = proj_dim
+        self.device = device
+
+    def forward(self, audio: Union[List[np.array], np.array]) -> torch.Tensor:
+        pass
+
+
+class WhisperEncoder(BaseAudioEncoder):
+    def __init__(
+            self,
+            proj_dim: int,
+            model_dim: int = 384,
+            checkpoint: str = "openai/whisper-tiny",
+            device: str = "cpu"
+    ):
+        super().__init__(proj_dim, device)
+        self.model = WhisperModel.from_pretrained(checkpoint).encoder.to(self.device)
+        self.extractor = WhisperFeatureExtractor.from_pretrained(checkpoint)
+        self.projection_layer = torch.nn.Linear(model_dim, self.proj_dim).ti(self.device)
```

Comment:
ti? 😸

Reply:
Lol, that happened because I added .to(device) to WhisperEncoder after i have checked